esp-hal-common/src/spi.rs

```diff
@@ -4,14 +4,27 @@
 //! which way you choose, you must first create an SPI instance with
 //! [`Spi::new`].
 //!
-//! ```rust
+//! ```rust,no_run
+//! # #![no_std]
+//! # use esp_hal_common::{IO, clock::ClockControl, peripherals::Peripherals, prelude::*, spi::{Spi, SpiMode}};
+//! # fn main () {
+//! # let peripherals = Peripherals::take();
+//! # #[cfg(system)]
+//! # let mut system = peripherals.SYSTEM.split();
+//! # #[cfg(pcr)]
+//! # let mut system = peripherals.PCR.split();
+//! # #[cfg(dport)]
+//! # let mut system = peripherals.DPORT.split();
+//! # let mut peripheral_clock_control = system.peripheral_clock_control;
+//! # let mut clocks = ClockControl::boot_defaults(system.clock_control).freeze();
+//! # 
```

Comment:
```suggestion
//! #
```

`rustfmt` doesn't leave us any space.